addressing/README.md

```diff
@@ -0,0 +1,329 @@
+| `dnsaddr` | Resolves multiaddrs from a special TXT record.                     |
+
+
+When the `/dns` protocol is used, the lookup may result in both IPv4 and IPv6
+addresses, in which case IPv6 will be preferred. To explicitly resolve to IPv4
+or IPv6 addresses, use the `/dns4` or `/dns6` protocols, respectively. 
+
+Note that in some restricted environments, such as inside a web browser, libp2p
+may not have access to the resolved IP addresses at all, in which case the
+runtime will determine what IP version is used.
+
+When a name-based multiaddr encapsulates another multiaddr, only the name-based
+component is affected by the lookup process. For example, if `example.com`
+resolves to `1.2.3.4`, libp2p will resolve the address
+`/dns4/example.com/tcp/42` to `/ip4/1.2.3.4/tcp/42`.
+
+#### dnsaddr Links
+
+A libp2p-specific DNS-backed format, `/dnsaddr` resolves addresses from a `TXT`
+record associated with the `_dnsaddr` subdomain of a given domain.
+
+For example, resolving `/dnsaddr/libp2p.io` will perform a `TXT` lookup for
+`_dnsaddr.libp2p.io`. If the result contains an entry of the form
+`dnsaddr=<multiaddr>`, the embedded multiaddr will be parsed and used.
+
+The `dnsaddr` lookup serves a similar purpose to a standard A-record DNS lookup,
+however there are differences that can be important for some use cases. The most
+significant is that the `dnsaddr` entry contains a full multiaddr, which may
+include a port number or other information that an A-record lacks, and it may
+even specify a non-IP transport. Also, there are cases in which the A-record
+already serves a useful purpose; using `dnsaddr` allows a second "namespace" for
+libp2p registrations.
+
+### TCP
+
+The libp2p TCP transport is supported in all implementations and can be used
+wherever TCP/IP sockets are accessible.
+
+Addresses for the TCP transport are of the form `<ip-multiaddr>/tcp/<tcp-port>`,
+where `<ip-multiaddr>` is a multiaddr that resolves to an IP address, as
+described in the [IP and Name Resolution section](#ip-and-name-resolution). 
+The `<tcp-port>` argument must be a 16-bit unsigned integer.
+
+### WebSockets
+
+WebSocket connections are encapsulated within TCP/IP sockets, and the WebSocket
+multiaddr format mirrors this arrangement.
+
+A libp2p WebSocket multiaddr is of the form `<tcp-multiaddr>/ws`, where
+`<tcp-multiaddr`> is a valid mulitaddr for the TCP transport, as [described
+above](#tcp).
+
+### QUIC
+
+QUIC sessions are encapsulated within UDP datagrams, and the libp2p QUIC
+multiaddr format mirrors this arrangement.
+
+A libp2p QUIC multiaddr is of the form `<ip-multiaddr>/udp/<udp-port>/quic`,
+where `<ip-multiaddr>` is a multiaddr that resolves to an IP address, as
+described in the [IP and Name Resolution section](#ip-and-name-resolution). 
+The `<udp-port>` argument must be a 16-bit unsigned integer.
+
+
+### `p2p-circuit` Relay Addresses
+
+The libp2p [circuit relay protocol][relay-spec] allows a libp2p peer to relay
```

Comment:
I'd frame this description slightly differently, from the perspective of the endpoints being connected:

The libp2p [circuit relay protocol][relay-spec] allows a libp2p peer A to communicate with another peer B via a third party C. This is useful for circumstances where A and B would be unable to communicate directly.

Reply:
Thank you for the suggestion, applied in 90e852d.